### ISSUE TITLE
No longer developing PaperWM. Looking for new maintainers moving forward.

### ISSUE THREAD

**Issue by jtaala** [paperwm/PaperWM#980]
Evening all,

Please see #945 for some history here.

Unfortunately, I haven't found anyone to take over PaperWM development and/or continue to maintain it.

As such, ~~I'm planning on archiving this repo in a month or so~~.

Please reach out if you're keen to take over ownership of PaperWM!

**Comment by jtaala** [comment]
@smichel17, @Thesola10 - let me know if you (or anyone you know) might be keen to take over PaperWM development (or at least ownership of this repo).

**Comment by capfredf** [comment]
I just came across the awesome extension. I am still playing with it. Not sure if I am capable of maintaining it and developing features for now. If no body is willing to step in, I can take it over (at least the ownership of this repo)

**Comment by smichel17** [comment]
My situation is pretty similar to when @jtaala joined as maintainer:

- I don't have much time/inclination to be a maintainer myself (I lack the knowledge of gnome internals and don't want to invest the time to learn).
    - There's a small chance I'll have more bandwidth in around 6 months. That would probably look like doing code reviews. But that's a while off and I don't know exactly how things will go, so don't count on it :)
- I am willing to help steward a new maintainer joining. Which essentially means being a trustworthy person to manage permissions, so if we add a new maintainer and it turns into an xz situation, I can regain control of the repo.

**Comment by smichel17** [comment]
Just noticed this part.

> As such, I'm planning on archiving this repo in a month or so.

I think PaperWM works well enough for most people who use it today, so new maintainers are (only) going to show up when someone using the app upgrades to an unsupported gnome version and finds PaperWM hasn't been upgraded. So I'd rather leave the repo open until that happens. If nobody shows up to take over once there's a gnome version with breaking changes, then we can archive it, but I'd vote to wait until then. The repo was open with minimal maintenance for a full year before you took over.

**Comment by jtaala** [comment]
Thanks @smichel17 - but we do need an admin or someone to take control of the repo.  My situation is that I won't be available to administer/control/maintain PaperWM.  Without an admin (who can add/remove contributors etc.) I'd want to archive this project (at least until a new repo owner/admin can be found).

Essentially, we at least need a new repo admin, who could: 
- maintain the repo, 
- add/remove contributors, 
- respond to bug/feature requests/ questions etc., 
- review/accept PRs (and merge changes into `release`), 
- eventually also give someone else these admin perms;
 
I'll also need to migrate the current extensions.gnome.org account to this new person so that new version of PaperWM can be deployed.

**Comment by jtaala** [comment]
> * Which essentially means being a trustworthy person to manage permissions, so if we add a new maintainer and it turns into an xz situation, I can regain control of the repo.

Indeed, this person should be very familiar with the project, trustworthy, etc.  

Let me know if you'd be open to become a repo admin here (essentially to be able to review/accept PRs, add/remove contributors, maintain repo perms etc.).

Sounds like you're situation might be similar to mine though.

**Comment by jtaala** [comment]
> Just noticed this part.
> 
> > As such, I'm planning on archiving this repo in a month or so.
> 
> I think PaperWM works well enough for most people who use it today, so new maintainers are (only) going to show up when someone using the app upgrades to an unsupported gnome version and finds PaperWM hasn't been upgraded. So I'd rather leave the repo open until that happens. If nobody shows up to take over once there's a gnome version with breaking changes, then we can archive it, but I'd vote to wait until then. The repo was open with minimal maintenance for years before you took over.

Sorry @smichel17!

I thought I was the only admin left (I hadn't seen you in a long time so didn't recognise the name!).

Indeed, you are also an admin (as far as I can see).  In this case - if you're around and can do basic repo maintenance and respond to questions, add new contributors etc. we could keep it open until breaking changes (i.e. I'd be happy to not archive the project).

My concerns here though are that without someone to respond to new questions/bugs/feature requests, or PRs etc. we'll essentially be in an unmaintained state.  And archiving the project - until someone that can be more involved is more available, I would have thought would be preferable (otherwise, questions, PRs etc. would go unanswered etc.) - and would likely remain as such.

Users could always fork, or continue to use the last release etc. (even if archived).

**Comment by smichel17** [comment]
> I hadn't seen you in a long time so didn't recognise the name

Yeah, the history is in https://github.com/paperwm/PaperWM/issues/407. I haven't been as active in FLOSS as I would like the past 3 years, since I got my current job, but that will hopefully change in around 6 months.

I have access to the [organization](https://github.com/paperwm), not just the repo, so another repo admin can't remove my access; only `@hedning` or `@olejorgenb` can do that. That makes it safe for me to give out to someone else without worrying too much about a takeover from a bad actor (e.g. SEO spammer).

I think it's OK if nobody responds to questions/bugs/feature requests. For PRs.. well, if we get enough of those, maybe the people sending them can do some review, too -- they are the potential future maintainers :). We could even have a policy like: if you want your PR merged, you need to review someone else's PR first.

Forking is a lot more effort than taking over maintainership, because you lose the community; So you also need to do the work of telling everybody about the fork (competing with any other forks, and folks who are just staying with the archived repo).

I think it's fine for you to just step back; you can even remove yourself as admin if you don't want to get pinged. I'll make sure it's possible for someone else to step up as a new maintainer if they show interest.

**Comment by Thesola10** [comment]
I am interested in taking up the role of interim repo admin (or co-admin alongside @smichel17). I am currently focusing on my studies, so this will be a best-effort thing, but I will, at least, respond to and review PRs, and clean up issues from time to time.

It would pain me greatly to see PaperWM archived and falling into slumber again, so I am at least willing to help prevent the PaperWM-community split from happening again.

If anything, PaperWM-community is proof that many people (like me!) like PaperWM enough to step up and help fix it for the latest GNOME, even if the admins are unresponsive.

So please, don't archive the repository, it would likely be a death knell for what I think is a beautiful desktop paradigm.

Thank you again @jtaala for all that you have done.

**Comment by jtaala** [comment]
> It would pain me greatly to see PaperWM archived and falling into slumber again, so I am at least willing to help prevent the PaperWM-community split from happening again.
> 
> If anything, PaperWM-community is proof that many people (like me!) like PaperWM enough to step up and help fix it for the latest GNOME, even if the admins are unresponsive.
> 
> So please, don't archive the repository, it would likely be a death knell for what I think is a beautiful desktop paradigm.

That's awesome @Thesola10!

Yes, we won't be archiving PaperWM!  This is indeed proof (and what I was hoping to see) that others in the PaperWM community are able to help keep PaperWM alive and evolving!

Your PRs have been great @Thesola10.  If there's at least a few that can give best efforts (when they are able to), then there's no reason to archive/pause PaperWM.  

TBH, as of late (due to life, work, family, and other responsibilities) I just haven't been able to continue doing releases, let alone look at PRs, fixing bugs, and answering questions.  

Given I (thought) I was the only one able to do this, and having invested considerable effort in rewriting much of PaperWM (and keeping it up to date with Gnome etc.) - I felt it better (at the time)) to look at archiving until we had a few others that could keep it going.  The responses here have definitely proved that!

I'm trying to be involved as much as I can - but this particular coming year looks to pretty hectic for me, so I couldn't commit as much as I would want.

I'm hoping in the future I can get back to a place where I'm a bit free-er to start contributing lots more.

**Comment by NMartin354** [comment]
Is this closed, reading through it, I'm kind of re-vitalized if you're not interested.

**Comment by jtaala** [comment]
> Is this closed, reading through it, I'm kind of re-vitalized if you're not interested.

Definitely welcome to jump in and start contributing @NMartin354 and we can add you as collaborator very quickly.  Main issue was having someone available to admin the repo, do releases as others contibute.  @Thesola10 is stepping in for some of this.

**Comment by jtaala** [comment]
@Thesola10 - I'll reach out to Javad (@JustPerfection) who does the EGO approvals for new versions.  See if he can just add you over paperwm so you can do releases on EGO from your account.

**Comment by jtaala** [comment]
> [@Thesola10](https://github.com/Thesola10) - I'll reach out to Javad (@JustPerfection) who does the EGO approvals for new versions. See if he can just add you over paperwm so you can do releases on EGO from your account.

![Image](https://github.com/user-attachments/assets/3c73f66c-e9ec-4c62-8de2-b90b62e13cac)

No way to do this (other than to share password it seems).  @Thesola10 I'll send you a link to create an account on my vaultwarden instance to share (with others) the EGO password.

**Comment by jtaala** [comment]
Just a note @Thesola10, I sent you an invite last week (to your email address) to the paperwm vaultwarden instance which allows you to access the extensions.gnome.org password for the paperwm account.

**Comment by Thesola10** [comment]
Thanks @jtaala, for some reason I only received the vaultwarden mail today.

**Comment by jtaala** [comment]
> Thanks [@jtaala](https://github.com/jtaala), for some reason I only received the vaultwarden mail today.

All good - I just clicked the "resend invite" button today (a few minutes ago).

You can always check/login there to see any changes to the paperwm ego account.

**Comment by jtaala** [comment]
Super easy to update the version on ego.  Just generate the zip by running the `generate-extension-zip.sh` script and upload the resultant zip to https://extensions.gnome.org/upload/

**Comment by malcolmredheron** [comment]
Oh dear! I came to the PaperWM github page to look for or file an issue and found this. PaperWM is by far my favorite Gnome extension and I'm highly motivated to ensure that it continues to thrive. I have too little spare time to be excited about learning about Gnome internals, but if there are other ways that I help, including by donating money, I'd be very glad to do that.. When I discovered PaperWM, I was frustrated that I could only "buy a coffee" instead of setting up recurring donations, which was what I wanted to do

**Comment by smichel17** [comment]
I'll unpin this. I think it's handled well enough. We can use more maintainers, but that's pretty much always the case for volunteer FLOSS projects like this.
